### PR TITLE
Fix generics being parsed wrong in idl

### LINKED ISSUE #3013: bug: Generic structs error in the idl if they're populated with custom structs
The title says it all - if we do:

```
#[account]
struct MyAccount {
    data: GStruct<u32>,
}

#[derive(AnchorSerialize, AnchorDeserialize, Clone)]
struct GStruct<T> {
    data: T,
}
```

the generated idl is fine and can be used. If we do

```
#[account]
struct MyAccount {
    data: GStruct<MyStruct>,
}

#[derive(AnchorSerialize, AnchorDeserialize, Clone)]
struct GStruct<T> {
    data: T,
}

#[derive(AnchorSerialize, AnchorDeserialize, Clone)]
struct MyStruct {
    data: u32,
}
```

the generated idl is unparsable. Might open a PR for this later, but in case I forget wanted to at least have the issue. MRE for this here: https://github.com/cryptopapi997/anchor-generics-mre

### PR DESCRIPTION
**Pull request by cryptopapi997**
Resolves https://github.com/coral-xyz/anchor/issues/3013

### CODE REVIEW

File: ts/packages/anchor/src/coder/borsh/idl.ts
Comment (acheroncrypto) [inline ts/packages/anchor/src/coder/borsh/idl.ts:1]: There are [formatting errors](https://github.com/coral-xyz/anchor/actions/runs/9450122704/job/26027976172#step:15:24) in this file. Could you run `yarn lint:fix`?

File: ts/packages/anchor/tests/coder-types.spec.ts
Comment (acheroncrypto) [inline ts/packages/anchor/tests/coder-types.spec.ts:1]: Thanks for adding a test case, but there is a similar test here

https://github.com/coral-xyz/anchor/blob/d6e43b6312327626a698cdf2bc300069298b7b91/tests/idl/programs/new-idl/src/lib.rs#L110

https://github.com/coral-xyz/anchor/blob/d6e43b6312327626a698cdf2bc300069298b7b91/tests/idl/tests/new-idl.ts#L335

Could you move the test there too?

**Review by acheroncrypto — CHANGES_REQUESTED** [review]

File: ts/packages/anchor/tests/coder-types.spec.ts
Comment (cryptopapi997) [reply ts/packages/anchor/tests/coder-types.spec.ts:1]: ah, didn't know ab this. Done!

File: ts/packages/anchor/src/coder/borsh/idl.ts
Comment (cryptopapi997) [reply ts/packages/anchor/src/coder/borsh/idl.ts:1]: done!

**Review by acheroncrypto — APPROVED** [review]
These one line fixes are the best, thanks!

**Pull request merged**